File: Geek Assistant (Android AIO)/Modules/General/Action/GA_PrepareAppdata.vb
Imports System.IO

Module GA_PrepareAppdata
    Public Sub Run()
        If Not Directory.Exists(GA) Then _
            Directory.CreateDirectory(GA)
        If Not Directory.Exists(GA_tools) Then _
            Directory.CreateDirectory(GA_tools)
        If Not File.Exists($"{GA_tools}\adb.exe") Then _
            File.WriteAllBytes(GA_tools & "\adb.exe", My.Resources.adb)
        'If Not File.Exists(GA_tools & "\adbUniversalDrivers-byGoogle.exe") Then File.WriteAllBytes(GA_tools & "\adbUniversalDrivers-byGoogle.exe", My.Resources.ces.adbUniversalDrivers_byGoogle)
        If Not File.Exists($"{GA_tools}\AdbWinApi.dll") Then _
            File.WriteAllBytes(GA_tools & "\AdbWinApi.dll", My.Resources.AdbWinApi)
        If Not File.Exists($"{GA_tools}\AdbWinUsbApi.dll") Then _
            File.WriteAllBytes(GA_tools & "\AdbWinUsbApi.dll", My.Resources.AdbWinUsbApi)
        If Not File.Exists($"{GA_tools}\fastboot.exe") Then _
            File.WriteAllBytes(GA_tools & "\fastboot.exe", My.Resources.fastboot)
    End Sub
End Module

File: Geek Assistant (Android AIO)/Modules/General/Action/GA_PrepareAppdata.cs
using System.IO;

internal static partial class GA_PrepareAppdata {
    public static void Run() {
        if (!Directory.Exists(common.GA))
            Directory.CreateDirectory(common.GA);
        if (!Directory.Exists(common.GA_tools))
            Directory.CreateDirectory(common.GA_tools);
        if (!File.Exists($@"{common.GA_tools}\adb.exe"))
            File.WriteAllBytes($@"{common.GA_tools}\adb.exe", prop.tools.adb);
        if (!File.Exists($@"{common.GA_tools}\AdbWinApi.dll"))
            File.WriteAllBytes($@"{common.GA_tools}\AdbWinApi.dll", prop.tools.AdbWinApi);
        if (!File.Exists($@"{common.GA_tools}\AdbWinUsbApi.dll"))
            File.WriteAllBytes($@"{common.GA_tools}\AdbWinUsbApi.dll", prop.tools.AdbWinUsbApi);
        if (!File.Exists($@"{common.GA_tools}\fastboot.exe"))
            File.WriteAllBytes($@"{common.GA_tools}\fastboot.exe", prop.tools.fastboot);
        if (!File.Exists($@"{common.GA_tools}\busybox"))
            File.WriteAllBytes($@"{common.GA_tools}\busybox", prop.tools.busybox);
    }
}